registry/storage/driver/storagedriver.go

```diff
@@ -106,11 +106,11 @@ type FileWriter interface {
 }
 
 // PathRegexp is the regular expression which each file path must match. A
-// file path is absolute, beginning with a path components and may contain
-// more of them separated by slashes, where each path component is
+// file path is absolute, beginning with a slash or a path components and may
+// contain more of them separated by slashes, where each path component is
 // restricted to alphanumeric characters or a period, underscore, colon or
 // hyphen.
-var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]+(/[A-Za-z0-9._:-]+)*)+$`)
+var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]*(/[A-Za-z0-9._:-]+)*)+$`)
```

Comment:
i think i am trying to do exactly what you said is a correct definition of path which are typically accessed as sub paths...
Current master or latest release mandates the path to start with a `/` https://github.com/docker/distribution/blob/master/registry/storage/driver/storagedriver.go#L113
whereas my regex allows to specify sub paths like `asdf/qwer`
This fix is top of another fix which i had done https://github.com/deis/distribution/pull/7/files#diff-f67a662acfd4aa13eaf56b5bb38f3421R109